include/envoy/ssl/connection.h

```diff
@@ -148,9 +149,16 @@ class ConnectionInfo {
    * exists.
    */
   virtual absl::optional<std::string> x509Extension(absl::string_view extension_name) const PURE;
+
+  /**
+   * Performs a TLS handshake on the SSL object and returns an action indicating
+   * whether the callsite should close the connection or keep it open.
+   */
+  virtual Network::PostIoAction doHandshake() PURE;
```

Comment:
My didn't meant this to be merged in a single interface into ConnectionInfo. This should still in a Handshaker interface. While HandshakerImpl should implement both Handshake and ConnectionInfo.

Reply:
OK, I created a separate handshaker interface. For now, SslSocketInfo implements both Handshaker and ConnectionInfo.